torch/testing/_internal/common_methods_invocations.py

```diff
@@ -280,6 +311,60 @@ def __init__(self,
         #   outside a function's domain.
         self._domain_eps = 1e-5
 
+    def sample_sparse_inputs(self, device, dtype, requires_grad=False, **kwargs):
+        """Returns an iterable of SampleInputs."""
+        if not self.supports_sparse:
+            raise RuntimeError("This op does not support sparse input")
+
+        low, high = self.domain
+        low = -10 if low is None else low + self._domain_eps
+        high = 10 if high is None else high - self._domain_eps
+        is_coalesced = kwargs.get("is_coalesced", False)
```

Comment:
I would expect sparse sample inputs for the unary ufuncs to generate a variety of tensors, including coalesce, uncoalesced and hybrid tensors. Is it really interesting to continue separating them? I always though that separation in the current sparse test suite was an artifact of how the tests are generated.

Put another way: the sample inputs are supposed to be the sample inputs, we shouldn't intend to partition them based on different kwargs.

Reply:
I agree. Let's put everything inside a single list. The reason for partition the input is that during the test we can see on which data the test is failing...

Reply:
I understand there's a tension between test granularity and this approach. I think we should solve the granularity issue differently however, by adding support for named subtests, for example, or naming the sample inputs and throwing more informative failure messages.